wallet/src/de/schildbach/wallet/ui/EncryptKeysDialogFragment.java

```diff
@@ -319,7 +324,10 @@ public void run() {
                                     log.info("wallet successfully decrypted");
                                 } catch (final KeyCrypterException x) {
                                     log.info("wallet decryption failed: " + x.getMessage());
-                                    pinRetryController.failedAttempt(oldPassword);
+                                    if(pinRetryController.failedAttempt(oldPassword)) {
+                                        restartService.performRestart(getActivity(), true);
+                                        dismiss();
+                                    }
```

Comment:
In theory this dialog is not used anymore due to the changes with to resolve the Non-encrypted Wallets crashes and errors.

We can get rid of it in a new story.

This dialog was the old Set Pin dialog.